File: Dw.ClickHouse/Dtos/Output/ChOutputDto.cs
using Ch.Models.Dtos.Condition;
using Ch.Models.Enums.Output;

namespace Ch.Models.Dtos.Output
{
    public class ChOutputDto : ChColumnOutputDto
    {
        public ChOutputDto SubqueryColumn { get; set; }

        public OutputAction OutputAction { get; set; } = OutputAction.ColumnAdd;
        public ChColumnOutputDto PartitionByColumn { get; set; }
        public List<ChColumnOutputDto> PartitionOverColumns { get; set; } = new List<ChColumnOutputDto>();
        public List<ChColumnOutputDto> SumColumns { get; set; } = new List<ChColumnOutputDto>();
        public AggregateFunction AggregateFunction { get; set; } = AggregateFunction.None;

        public List<ChConditionDto> Conditions { get; set; } = new List<ChConditionDto>();
    }
}

File: WebApplication/src/app-clickhouse/dtos/output/ch-output.dto.ts
import { AggregateFunction } from "../../enums/output/aggregate-function.enum";
import { ChColumnOutputDto } from "./ch-column-output.dto";
import { ChConditionDto } from "../condition/ch-condition.dto";
import { OutputAction } from "../../enums/output/output-action.enum";

export class ChOutputDto extends ChColumnOutputDto {
    subqueryColumn: ChOutputDto;

    outputAction = OutputAction.columnAdd;
    partitionByColumn: ChOutputDto;
    partitionOverColumns: ChColumnOutputDto[] = [];
    sumColumns: ChColumnOutputDto[] = [];
    aggregateFunction = AggregateFunction.none;

    conditions: ChConditionDto[] = [];

    // Client only
    rawSelect: string;
}